pkg/anonymization/builder_test.go

```diff
@@ -0,0 +1,133 @@
+			builder: getBuilderInstance(),
+		},
+		{
+			name:            "method 'AddSensitiveValue' sets values on the anonymizer instance",
+			builder:         getBuilderInstance().AddSensitiveValue("mock", "xxxx"),
+			sensitiveValues: map[string]string{"mock": "xxxx"},
+		},
+		{
+			name:         "method 'WithConfigClient' sets the client on the anonymizer instance",
+			builder:      getBuilderInstance().WithConfigClient(fakeCfgClient),
+			configClient: fakeCfgClient,
+		},
+		{
+			name:         "method 'WithConfigurator' sets the configurator on the anonymizer instance",
+			builder:      getBuilderInstance().WithConfigurator(fakeConfigurator),
+			configurator: fakeConfigurator,
+		},
+		{
+			name:       "method 'WithDataPolicy' sets the policy on the anonymizer instance",
+			builder:    getBuilderInstance().WithDataPolicy(v1alpha1.ObfuscateNetworking),
+			dataPolicy: v1alpha1.ObfuscateNetworking,
+		},
+		{
+			name:       "method 'WithKubeClient' sets the client on the anonymizer instance",
+			builder:    getBuilderInstance().WithKubeClient(fakeKubeClient),
+			kubeClient: fakeKubeClient,
+		},
+		{
+			name:          "method 'WithNetworkClient' sets the client on the anonymizer instance",
+			builder:       getBuilderInstance().WithNetworkClient(fakeNetworkClient),
+			networkClient: fakeNetworkClient,
+		},
+		{
+			name:     "method 'WithNetworks' sets the networks on the anonymizer instance",
+			builder:  getBuilderInstance().WithNetworks([]string{"127.0.0.0/8", "192.168.0.0/16"}),
+			networks: []string{"127.0.0.0/8", "192.168.0.0/16"},
+		},
+		{
+			name:             "method 'WithRunningInCluster' sets the value on the anonymizer instance",
+			builder:          getBuilderInstance().WithRunningInCluster(true),
+			runningInCluster: true,
+		},
+		{
+			name:          "method 'WithSecretsClient' sets the client on the anonymizer instance",
+			builder:       getBuilderInstance().WithSecretsClient(fakeSecretsClient),
+			secretsClient: fakeSecretsClient,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// When
+			test, err := tc.builder.Build()
+
+			// Assert
+			assert.NoError(t, err)
+			assert.IsType(t, Anonymizer{}, *test)
+			if tc.sensitiveValues != nil {
+				assert.EqualValues(t, tc.sensitiveValues, test.sensitiveValues)
+			}
+			if tc.configClient != nil {
+				assert.Equal(t, tc.configClient, test.configClient)
+			}
+			if tc.configurator != nil {
+				assert.Equal(t, tc.configurator, test.configurator)
+			}
+			if tc.dataPolicy != "" {
+				assert.Equal(t, tc.dataPolicy, test.dataPolicy)
+			}
+			if tc.kubeClient != nil {
+				assert.Equal(t, tc.kubeClient, test.gatherKubeClient)
+			}
+			if tc.networkClient != nil {
+				assert.Equal(t, tc.networkClient, test.networkClient)
+			}
+			if len(tc.networks) > 0 {
+				assert.Equal(t, len(tc.networks), len(test.networks))
+			}
+			if tc.runningInCluster {
```

Comment:
I think only this condition is really required :)

Reply:
Solved in bc66284efa66957e06efbc3efd2c1f893793085e, thanks for the review